### ISSUE TITLE
[Bug] Ascend NPU: RMSNorm crashes with elementwise_affine=False; _native_npu FA rejects [B, N, 1, Skv] masks (LTX-2)

### ISSUE THREAD

**Issue by mengchengTang** [huggingface/diffusers#14380]
### Describe the bug

When running LTX-2 with `attn_backend=_native_npu` on Ascend NPU, two incompatibilities show up:

1. **RMSNorm** – layers with `elementwise_affine=False` leave `weight=None`, but `torch_npu.npu_rms_norm` requires a gamma tensor → crash (`gamma is None`).

2. **Fused attention mask** – LTX cross-attn uses masks shaped `[B, N, 1, Skv]` (e.g. `[1, 32, 1, 1024]`). Ascend FA does not broadcast the singleton query-length dim the way SDPA does, so `_maybe_modify_attn_mask_npu` must expand it to `[B, N, Sq, Skv]`. Today only `[B, 1, 1, Skv]` is expanded.

These bugs were discovered while running LTX-2.3 text-to-audio-video FlowGRPO training on Ascend NPU using [verl-omni](https://github.com/verl-project/verl-omni/blob/main/examples/flowgrpo_trainer/ltx2/run_ltx2_3_t2av_lora_npu.sh).

### Reproduction

Requires Ascend NPU + `torch_npu`. Minimal sketches of both failure modes:

```python
import torch
from diffusers.models.normalization import RMSNorm
from diffusers.models.attention_dispatch import (
    AttentionBackendName,
    attention_backend,
    dispatch_attention_fn,
)

# --- Bug 1: RMSNorm with elementwise_affine=False ---
norm = RMSNorm(dim=64, eps=1e-6, elementwise_affine=False).to("npu")
x = torch.randn(2, 16, 64, device="npu", dtype=torch.float16)
# Crashes: npu_rms_norm called with weight=None
y = norm(x)

# --- Bug 2: FA mask [B, N, 1, Skv] under _native_npu ---
B, Sq, Skv, N, D = 1, 384, 1024, 32, 64
q = torch.randn(B, Sq, N, D, device="npu", dtype=torch.float16)
k = torch.randn(B, Skv, N, D, device="npu", dtype=torch.float16)
v = torch.randn(B, Skv, N, D, device="npu", dtype=torch.float16)
attn_mask = torch.zeros(B, N, 1, Skv, device="npu", dtype=torch.float16)  # LTX-style

with attention_backend(AttentionBackendName._NATIVE_NPU):
    # Fails: Ascend FA expects Sq on dim=-2, not a singleton 1
    out = dispatch_attention_fn(q, k, v, attn_mask=attn_mask)
```

### Logs

# Error 1

<img width="873" height="478" alt="Image" src="https://github.com/user-attachments/assets/d3122afa-7dac-45ed-a653-9f5e0cda7b4c" />

# Error 2

<img width="873" height="478" alt="Image" src="https://github.com/user-attachments/assets/356c803b-2d7c-4f6f-9d3b-f4d04e2a79ad" />

### System Info

- OS: Linux aarch64
- Hardware: Ascend NPU
- Python: 3.11
- torch: 2.10.0
- torch_npu: 2.10.0
- CANN: 9.0.0
- diffusers: main

### Who can help?

@yiyixuxu @sayakpaul

**Comment by mengchengTang** [comment]
I have a fix ready in https://github.com/huggingface/diffusers/pull/14288

**Comment by zhtmike** [comment]
I think it is better to state clearly which model you are trying to fix on NPU. The inference result before and after this fixing (better with some code snippet), the related torch version and torch npu version. So diffusers committer can understand the problem clearly.

**Comment by mengchengTang** [comment]
@zhtmike Thanks for the suggestion. Let me make the model and validation scope more explicit.

The affected model is `dg845/LTX-2.3-Diffusers`, used for LTX-2.3 text-to-audio-video LoRA FlowGRPO training in verl-omni.

Environment:
- Python 3.11
- PyTorch 2.10.0
- torch-npu 2.10.0
- CANN 9.0.0
- diffusers `main`
- Ascend NPU, Linux aarch64

I verified the fix in two ways:

1. Using both minimal scripts from the Reproduction section:
   - Before the fix, `RMSNorm(elementwise_affine=False)` passes `weight=None` to `npu_rms_norm` and crashes. After the fix, it falls back to the existing PyTorch RMSNorm implementation and completes successfully.
   - Before the fix, the LTX cross-attention mask `[B, N, 1, Skv]` is rejected by Ascend fused attention. After expanding it to `[B, N, Sq, Skv]`, the attention call completes successfully.

2. Using the end-to-end verl-omni LTX-2.3 recipe:
   `examples/flowgrpo_trainer/ltx2/run_ltx2_3_t2av_lora_npu.sh`

   This recipe uses the `dg845/LTX-2.3-Diffusers` checkpoint. Before the fix, execution stops at the RMSNorm or fused-attention mask errors described above. With the fix, the rollout/training path proceeds through both affected code paths without these errors.

This is a runtime correctness and NPU compatibility fix rather than an output-quality change. Before the fix, no inference output is produced because execution crashes; after the fix, the same workloads complete the affected operations successfully.

I will also add these before/after validation details to the issue/PR description so they are easier for maintainers to find.
